### PR TITLE
F/custom navmesh

### LINKED ISSUE #17: NPC Pathfinding
* implement A* logic

** Uses the navmesh in issue #19

### LINKED ISSUE #19: Generate map navmesh
* Procedurally generate the navmesh to follow the blocks

### PR DESCRIPTION
**Pull request by f-michaut**
Created the custom navmesh and a Astar function to use it and get a path from 2 nodes in the graph.

To testthe navmesh, just run the game, pause it, and go in the Scene tab in the editor. If you have Gizmos enabled, you will see the nodes and the links between them.

I didn't test the Astar function, I just copied it from my assignement. The function is static so you can call it from anywhere, and just require 2 nodes. To get nodes from the graph, there is a function `NavigationGraph.GetClosestNode(Vector3 pos)` to get the closest node to a 3D position. You can ask for 2 random positions, and get the nodes, and call the function with them as start and end nodes.

### CODE REVIEW

File: Assets/Scripts/Level generation scripts/NavigationGraph.cs [hunk 1]
```diff
@@ -0,0 +1,331 @@
+﻿using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Node
+{
+    public Vector3 position;
+    public List<Link> links;
```
Comment (ClaudiaLapalme) [inline Assets/Scripts/Level generation scripts/NavigationGraph.cs:8]: These two variables could probably be private and provide getters and setters. I don't see them being called outside of this class very often

File: Assets/Scripts/Level generation scripts/NavigationGraph.cs [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    }
+}
+
+public class Link
+{
+    public Node node;
+    public float weight;
+}
+
+public class NavigationGraph
```
Comment (ClaudiaLapalme) [inline Assets/Scripts/Level generation scripts/NavigationGraph.cs:129]: this class could probably be static since all of its functions are

**Review by ClaudiaLapalme — APPROVED** [review]
I have minor comments that I don't think are enough to block the PR from being merged. Good job!

**Review by tedgag — APPROVED** [review]
As discussed on Discord, we will test the pathfinding in another PR and make optimization to the graph algorithm if needed, but for now it works as expected

**Pull request merged**